### PR TITLE
For NCEP regtests, update to spack-stack 1.6, add option for gnu compiler and add Hercules 

### LINKED ISSUE #1139: Add NCEP regression test support for gnu and hercules
**Is your feature request related to a problem? Please describe.**
Take advantage of spack-stack installs on additional RDHPCS platforms/compilers 

**Describe the solution you'd like**
Update run_cmake_matrix to be able to use gnu or intel compiler + add support for hercules. 

**Describe alternatives you've considered**
n/a

**Additional context**
Gnu on hercules pointed out some issues with the ufs-weather-model, so this can take advantage of potentially finding additional uninitialized variables which is a common issue that is encountered. 

@MatthewMasarik-NOAA is going to provide METIS library installs on hera with gnu (completed) and on hercules for gnu and intel.

### LINKED ISSUE #1205: Update NCEP regtest modules for Rocky8 transition on hera
**Is your feature request related to a problem? Please describe.**
Need to update modules to use rocky8 builds on hera.  Will also use this opportunity to update to the newest spack-stack 1.6.0 

**Describe the solution you'd like**
Updates to regression tests to run on rocky8 nodes. 

**Describe alternatives you've considered**
none. 

**Additional context**
Will need intel (and ideally gnu) builds of METIS on hera for rocky8.

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
# Pull Request Summary
This PR adds an option for the capability to run intel or gnu compilers on hera and adds the new machine hercules as an option for regression testing.  Moreover, this PR updates hera to the rocky-8 notes and hera, orion and hercules to spack-stack 1.6.0.  

## Description
For regtests/bin/matrix_cmake_ncep 
There are now two inputs, the first being the path to the model directory and a second which is compiler for the regtests, intel (default) or gnu.   

@MatthewMasarik-NOAA created the installs for parMETIS on the new machines and updated the paths for consistency on the existing machines including moving us to spack-stack 1.6.0 and the rocky-8 nodes on hera. 

The transition to the Rocky-8 node exposed a very subtle uninitialized variable, which as been added here.  No answer changes are expected with this PR.  

Please also include the following information: 
* Add any suggestions for a reviewer: @MatthewMasarik-NOAA  
* Mention any labels that should be added:   _enhancement_
* Are answer changes expected from this PR?  None - just adding additional testing capability. 

### Issue(s) addressed

* Closes https://github.com/NOAA-EMC/WW3/issues/1139 
* Closes https://github.com/NOAA-EMC/WW3/issues/1205

### Commit Message
For NCEP regtests, add option for gnu compiler and new machine Hercules 
### Check list  

- [x] Branch is up to date with the authoritative repository (NOAA-EMC) develop branch. 
- [x] Checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop). 
- [ ] If a version number update is required, checked the [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number) checklist. 
- [ ] If a new feature was added, a regression test for testing the new feature is added. 
 
### Testing

* How were these changes tested?   On hera and orion, used intel to compare with develop.   On hera used gnu to compare with itself to create a baseline.  On hercules ran intel and gnu tests and compared with itself to create baselines.  
* Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) yes 
* Have the matrix regression tests been run (if yes, please note HPC and compiler)?  
** hera - intel (compare with develop) & gnu (compare with itself, no develop compare option) 
** orion - intel (compare with develop) 
** Hercules - intel (compare with instelf, no develop compare option) & gnu (compare with istelf, no develop compare option) 
* Please indicate the expected changes in the regression test output, (Note the [list](https://github.com/NOAA-EMC/WW3/wiki/How-to-use-matrix.comp-to-compare-regtests-with-develop#4-look-at-results) of known non-identical tests.) Should only be known b4b tests.  This is true of intel when comparing with develop (orion/hera) and hercules to itself (no develop baseline to compare with).  This is not yet true for gnu, but we will address these additional issues in subsequent issues/PRs. 
* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_): 

Updated output for comparisons for3/27/24: 

Hera compared with develop: 

```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR3_UQ_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (9 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (14 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (17 files differ)
mww3_test_03/./work_PR2_UQ_MPI_d2                     (14 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (16 files differ)
mww3_test_09/./work_MPI_ASCII                     (0 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (7 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
ww3_tp2.6/./work_ST4_ASCII                     (0 files differ)
ww3_ufs1.3/./work_a                     (3 files differ)
```
[heravDev.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14782362/heravDev.matrixCompFull.txt)
[heravDev.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14782363/heravDev.matrixCompSummary.txt)
[heravDev.matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/14782364/heravDev.matrixDiff.txt)

Hera intel vs itself: (Only expected non-b4b) 
[hera.intel.matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/14782372/hera.intel.matrixDiff.txt)
[hera.intel.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14782370/hera.intel.matrixCompFull.txt)
[hera.intel.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14782371/hera.intel.matrixCompSummary.txt)

Hera gnu vs istelf (no existing baseline for comparion) - this has more diffs than "expected" and will be addressed on subsequent PRs to resolve outstanding issues: 

[hera.gnu.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14782382/hera.gnu.matrixCompFull.txt)
[hera.gnu.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14782383/hera.gnu.matrixCompSummary.txt)
[hera.gnu.matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/14782384/hera.gnu.matrixDiff.txt)

Orion vs develop intel (only expected non b4b): 
[orionvdev.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14782388/orionvdev.matrixCompFull.txt)
[orionvdev.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14782389/orionvdev.matrixCompSummary.txt)
[orionvdev.matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/14782390/orionvdev.matrixDiff.txt)

Hercules intel vs itself (no existing basline for comparison, only expected non b4b): 
[hercules.intel.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14782391/hercules.intel.matrixCompFull.txt)
[hercules.intel.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14782392/hercules.intel.matrixCompSummary.txt)
[hercules.intel.matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/14782393/hercules.intel.matrixDiff.txt)

Hercules gnu vs itself (no existing baselines, some additional non b4b to be addressed in follow-up PRs): 

[hercules.gnu.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14782399/hercules.gnu.matrixCompFull.txt)
[hercules.gnu.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14782400/hercules.gnu.matrixCompSummary.txt)
(Diff file too big to post).

### CODE REVIEW

**Comment by MatthewMasarik-NOAA** [comment]
Thanks @JessicaMeixner-NOAA, I'll start the testing.

**Comment by MatthewMasarik-NOAA** [comment]
Fyi, @JessicaMeixner-NOAA on `orion` for test:  

`run_test -b slurm -o all -S -T -s MPI -s PDLIB -w work_mb -m grdset_b -f -p srun -n 24 ../model ww3_tp2.21`

I get a crash:
```
       grid           comp.      grd    pnt    trk    rst    bpt    prt
      -------------------------------------------------------------------
       glo_unst_b  00001-00024  00023  00022  -----  -----  -----  -----
      -------------------------------------------------------------------



  Initializations :
 --------------------------------------------------
       Model number  1 [glo_unst_b]
          Initializing wave model ...
forrtl: severe (24): end-of-file during read, unit -5, file Internal List-Directed Read
Image              PC                Routine            Line        Source
ww3_multi          0000000000875FC8  Unknown               Unknown  Unknown
ww3_multi          00000000008A3B22  Unknown               Unknown  Unknown
ww3_multi          00000000008A28A4  Unknown               Unknown  Unknown
ww3_multi          0000000000709DCE  w3uostmd_mp_load_         568  w3uostmd.F90
ww3_multi          000000000070D5AA  w3uostmd_mp_uost_         419  w3uostmd.F90
ww3_multi          00000000005BFFC9  w3iogrmd_mp_w3iog        1825  w3iogrmd.F90
ww3_multi          0000000000524411  w3initmd_mp_w3ini         728  w3initmd.F90
ww3_multi          0000000000442ACF  wminitmd_mp_wmini        2350  wminitmd.F90
ww3_multi          000000000041034F  MAIN__                    172  ww3_multi.F90
ww3_multi          00000000004101D2  Unknown               Unknown  Unknown
libc-2.17.so       00002ADEE2169495  __libc_start_main     Unknown  Unknown
ww3_multi          00000000004100E9  Unknown               Unknown  Unknown
```

I'll re-run this test to see if it goes away, though given it's SCOTCH on orion I'm a little concerned it could be an undiagnosed issue.

**Comment by JessicaMeixner-NOAA** [comment]
@MatthewMasarik-NOAA for this PR for orion only changed the location of the METIS install, so it shouldn't be this PR as the source of the issue.  But if this issue is reproducible it's definitely an issue we should look into.  I didn't see this in my tests: /work2/noaa/marine/jmeixner/PR_WW3/GNUPR/regtests/matrix13.out of this PR, but will be interested to see how the re-runs go on your end.

File: regtests/bin/matrix_cmake_ncep
Comment (MatthewMasarik-NOAA) [inline regtests/bin/matrix_cmake_ncep]: @JessicaMeixner-NOAA these are retained in for orion, but not present for hercules.  Also, their position for orion has been moved above the module loads.  Are these both intentional?

**Comment by MatthewMasarik-NOAA** [comment]
**Update**
Testing status for each platform / compiler:

* hera / intel - done. no issues.
* hera / gnu  - one set currently re-running that was run unintentionally w/ intel.
* orion / intel - re-run of ww3_tp2.21 successful.  just needs matrix.comp.
* hercules / gnu - errors in both sets.  needs troubleshooting.
* hercules / intel - errors in one set.

**Comment by JessicaMeixner-NOAA** [comment]
@MatthewMasarik-NOAA can you share more information offline about the Hercules errors?   The other updates are as I intended I'll do a double check on hercules environment variables though.

File: regtests/bin/matrix_cmake_ncep
Comment (JessicaMeixner-NOAA) [reply regtests/bin/matrix_cmake_ncep]: I actually don't see these being used in ufs-weather-model submit scripts.  I'm not sure why these are here although the `ulimit -s unlimited`  is usually needed.

Comment (MatthewMasarik-NOAA) [reply regtests/bin/matrix_cmake_ncep]: Those three were originally recommended on WCOSS2 when I was doing testing with SCOTCH a bit ago.  They were then also added to orion's job card when there were issues updating modules at a previous time and the issues seemed to go away.  I suspect we may need to include them on WCOSS2 when that update happens.  We have had all the `export`'s together at after the module loads, I don't know whether ordering matters with modules/exports, but I'd vote to choose one or the other to keep all job cards consistent between platforms 

I find it convenient to have the `cd` to regtests directory right after the `#SBATCH`s.  Could we keep that location?

Comment (JessicaMeixner-NOAA) [reply regtests/bin/matrix_cmake_ncep]: I'll look into this tomorrow.

Comment (JessicaMeixner-NOAA) [reply regtests/bin/matrix_cmake_ncep]: @MatthewMasarik-NOAA in my comparions  of what is happening in the develop branch on orion and what is happening in this branch in terms of the lodation of #sbatch and the "cd regtests" it is the same.  Is this different behavior for you?   

Since the hercules tests are working fine without extra envrionement variables beyond the unlimit -s, I prefer to not add anything.  We could add the environement variable to remove the one warning, but I'm okay leaving things as is.   I'll re-run a test on hercules to see if there are any system changes causing errors for me after my successful tests earlier this week/late last week.

**Comment by JessicaMeixner-NOAA** [comment]
During testing, @MatthewMasarik-NOAA discovered that we have an issue with omplace commands.  While an error message is seen on hercules+gnu, further inspection shows that this is likely an issue in develop as well.  See https://github.com/NOAA-EMC/WW3/issues/1152 for additional details.   This will be addressed before merging this PR.  Marking the PR as a draft until changes are made as requested by @MatthewMasarik-NOAA  and it's ready for re-review and testing.

File: regtests/bin/matrix_cmake_ncep
Comment (MatthewMasarik-NOAA) [reply regtests/bin/matrix_cmake_ncep]: @JessicaMeixner-NOAA, to tie up a loose end regarding the "cd regtests".  Checking now the job card for `hera`, I see it doesn't have either the extra `export`'s or those `ulimit` lines, so the "cd regtests" does come directly after the Slurm directives, as I had in mind.  However, for `orion`, these lines **are** currently above the "cd regtests", so I was wrong on that.  Since nothing changed, no fix needed.

Comment (JessicaMeixner-NOAA) [reply regtests/bin/matrix_cmake_ncep]: Thanks @MatthewMasarik-NOAA   

I have tested that we can remove the extra orion variables for our regression tests.  Orion is a bit unstable right now so we should re-test that when we have a resolution to https://github.com/NOAA-EMC/WW3/issues/1152 and are ready to re-test this PR.

**Comment by MatthewMasarik-NOAA** [comment]
> @MatthewMasarik-NOAA can you share more information offline about the Hercules errors? The other updates are as I intended I'll do a double check on hercules environment variables though.

@JessicaMeixner-NOAA and I did discuss this at the time.  I have some resolution on the hercules issues to update everyone.  @JessicaMeixner-NOAA found that the errors I had in hercules/intel were due to not downloading the regtest data from ftp, so that was resolved.  The hercules/gnu was interesting.  I did a second set that both had crashes like the first.  It seems that invoking conda setup in my .bashrc was the issue.  After disabling it both runs went through OK.

**Comment by JessicaMeixner-NOAA** [comment]
> > @MatthewMasarik-NOAA can you share more information offline about the Hercules errors? The other updates are as I intended I'll do a double check on hercules environment variables though.
> 
> @JessicaMeixner-NOAA and I did discuss this at the time. I have some resolution on the hercules issues to update everyone. @JessicaMeixner-NOAA found that the errors I had in hercules/intel were due to not downloading the regtest data from ftp, so that was resolved. The hercules/gnu was interesting. I did a second set that both had crashes like the first. It seems that invoking conda setup in my .bashrc was the issue. After disabling it both runs went through OK.

@MatthewMasarik-NOAA thanks for letting me know!   Once we have a resolution to https://github.com/NOAA-EMC/WW3/issues/1152 I'll re-test this PR and re open for review.

**Comment by JessicaMeixner-NOAA** [comment]
@MatthewMasarik-NOAA I've updated the title to reflect the additions of spack-stack 1.6.0.  While we still get some output from https://github.com/NOAA-EMC/WW3/issues/1152 we  confirmed that we are getting the expected performance of the OMP and think that we're doing okay as of now. Issue 1152 will remain open, but we will move forward with this PR, especially as we need to move to hera rock-8 nodes to continue testing other PRs.    My final set of tests results will be posted to this PR later this afternoon, but I think it's okay if you start to test this now as needing Rocky-8 on hera will hold up other PRs until we have that.

**Comment by JessicaMeixner-NOAA** [comment]
@MatthewMasarik-NOAA all logs are now posted.

**Comment by MatthewMasarik-NOAA** [comment]
> @MatthewMasarik-NOAA all logs are now posted.

Thanks @JessicaMeixner-NOAA

**Comment by JessicaMeixner-NOAA** [comment]
The follow up issue to track gnu issues can be found here: https://github.com/NOAA-EMC/WW3/issues/1146

**Comment by MatthewMasarik-NOAA** [comment]
Quick update to keep everyone informed.  The review for this PR is being finalized and will be posted shortly this morning.

**Review by MatthewMasarik-NOAA — APPROVED** [review]
## Code review
- Pass


## Testing
- Pass

### hera
#### intel:  develop v. fix
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR1_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (15 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (11 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (18 files differ)
mww3_test_03/./work_PR2_UQ_MPI_d2                     (15 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (12 files differ)
mww3_test_09/./work_MPI_ASCII                     (0 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (7 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
ww3_tp2.6/./work_ST4_ASCII                     (0 files differ)
ww3_ufs1.3/./work_a                     (3 files differ)
 
**********************************************************************
************************ identical cases *****************************
**********************************************************************
```
[hera.intel.dev.fix.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14840144/hera.intel.dev.fix.matrixCompSummary.txt)
[hera.intel.dev.fix.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14840143/hera.intel.dev.fix.matrixCompFull.txt)
[hera.intel.dev.fix.matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/14840145/hera.intel.dev.fix.matrixDiff.txt)


#### intel:  fix v. fix
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR3_UQ_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (11 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (12 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (17 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (18 files differ)
mww3_test_03/./work_PR2_UQ_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (16 files differ)
mww3_test_09/./work_MPI_ASCII                     (0 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (7 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
ww3_tp2.6/./work_ST4_ASCII                     (0 files differ)
ww3_ufs1.3/./work_a                     (3 files differ)
 
**********************************************************************
************************ identical cases *****************************
**********************************************************************
```
[hera.intel.fix.fix.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14840157/hera.intel.fix.fix.matrixCompSummary.txt)
[hera.intel.fix.fix.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14840156/hera.intel.fix.fix.matrixCompFull.txt)
[hera.intel.fix.fix.matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/14840158/hera.intel.fix.fix.matrixDiff.txt)



#### gnu:  fix v. fix
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR2_UNO_MPI_d2                     (17 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (13 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (12 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (14 files differ)
mww3_test_03/./work_PR2_UQ_MPI_d2                     (15 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (15 files differ)
mww3_test_09/./work_MPI_ASCII                     (0 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (7 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
ww3_tp2.17/./work_ma                     (1 files differ)
ww3_tp2.17/./work_a                     (1 files differ)
ww3_tp2.17/./work_mc1                     (1 files differ)
ww3_tp2.17/./work_mb                     (1 files differ)
ww3_tp2.17/./work_mc                     (1 files differ)
ww3_tp2.17/./work_ma1                     (1 files differ)
ww3_tp2.17/./work_c                     (1 files differ)
ww3_tp2.17/./work_b                     (1 files differ)
ww3_tp2.6/./work_ST4_ASCII                     (0 files differ)
ww3_ts1/./work_ST4_WRT                     (1 files differ)
ww3_ufs1.1/./work_c_npl                     (1 files differ)
ww3_ufs1.1/./work_d                     (1 files differ)
ww3_ufs1.1/./work_c_nth                     (1 files differ)
ww3_ufs1.1/./work_c                     (1 files differ)
ww3_ufs1.2/./work_a                     (4 files differ)
ww3_ufs1.2/./work_l                     (1 files differ)
ww3_ufs1.2/./work_c                     (3 files differ)
ww3_ufs1.2/./work_b                     (4 files differ)
ww3_ufs1.3/./work_a                     (29 files differ)
 
**********************************************************************
************************ identical cases *****************************
**********************************************************************
```
[hera.gnu.fix.fix.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14840194/hera.gnu.fix.fix.matrixCompSummary.txt)
[hera.gnu.fix.fix.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14840191/hera.gnu.fix.fix.matrixCompFull.txt)
[hera.gnu.fix.fix.matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/14840195/hera.gnu.fix.fix.matrixDiff.txt)


#### orion
##### intel:  develop v. fix
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR2_UQ_MPI_d2                     (14 files differ)
mww3_test_03/./work_PR2_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (15 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (16 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (8 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (18 files differ)
mww3_test_09/./work_MPI_ASCII                     (0 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (6 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
ww3_tp2.6/./work_ST4_ASCII                     (0 files differ)
ww3_ufs1.3/./work_a                     (3 files differ)
 
**********************************************************************
************************ identical cases *****************************
**********************************************************************
```
[orion.intel.dev.fix.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14840459/orion.intel.dev.fix.matrixCompSummary.txt)
[orion.intel.dev.fix.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14840458/orion.intel.dev.fix.matrixCompFull.txt)
[orion.intel.dev.fix.matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/14840460/orion.intel.dev.fix.matrixDiff.txt)


##### intel:  fix v. fix
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR2_UQ_MPI_d2                     (15 files differ)
mww3_test_03/./work_PR2_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (18 files differ)
mww3_test_03/./work_PR1_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (15 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (14 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (16 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (6 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (10 files differ)
mww3_test_09/./work_MPI_ASCII                     (0 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (7 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
ww3_tp2.6/./work_ST4_ASCII                     (0 files differ)
ww3_ufs1.3/./work_a                     (3 files differ)
 
**********************************************************************
************************ identical cases *****************************
**********************************************************************
```
[orion.intel.fix.fix.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14840648/orion.intel.fix.fix.matrixCompFull.txt)
[orion.intel.fix.fix.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14840650/orion.intel.fix.fix.matrixCompSummary.txt)
[orion.intel.fix.fix.matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/14840652/orion.intel.fix.fix.matrixDiff.txt)


#### hercules
##### intel:  fix v. fix
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR2_UQ_MPI_d2                     (15 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (17 files differ)
mww3_test_03/./work_PR1_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (17 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (13 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (12 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (14 files differ)
mww3_test_09/./work_MPI_ASCII                     (0 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (5 files differ)
ww3_tp2.6/./work_ST4_ASCII                     (0 files differ)
ww3_ufs1.3/./work_a                     (3 files differ)
 
**********************************************************************
************************ identical cases *****************************
**********************************************************************
```
[hercules.intel.fix.fix.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14840750/hercules.intel.fix.fix.matrixCompSummary.txt)
[hercules.intel.fix.fix.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14840749/hercules.intel.fix.fix.matrixCompFull.txt)
[hercules.intel.fix.fix.matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/14840752/hercules.intel.fix.fix.matrixDiff.txt)


##### gnu:  fix v. fix
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR2_UQ_MPI_d2                     (15 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (12 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (12 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (13 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (15 files differ)
mww3_test_07/./work_PR3_UQ                     (1 files differ)
mww3_test_09/./work_MPI_ASCII                     (0 files differ)
ww3_tic1.4/./work_IC0IS2_1000                     (1 files differ)
ww3_tic1.4/./work_IC1IS2_1000                     (1 files differ)
ww3_tp2.21/./work_ma                     (1 files differ)
ww3_tp2.21/./work_b_metis                     (1 files differ)
ww3_tp2.21/./work_a                     (1 files differ)
ww3_tp2.21/./work_b                     (1 files differ)
ww3_tp2.6/./work_ST0                     (1 files differ)
ww3_tp2.6/./work_pdlib                     (1 files differ)
ww3_tp2.6/./work_ST4                     (1 files differ)
ww3_tp2.6/./work_ST4_ASCII                     (2 files differ)
ww3_tp2.7/./work_ST0                     (1 files differ)
ww3_ufs1.1/./work_unstr_c                     (1 files differ)
ww3_ufs1.1/./work_unstr_b                     (1 files differ)
ww3_ufs1.1/./work_unstr_a                     (1 files differ)
ww3_ufs1.3/./work_a                     (3 files differ)
 
**********************************************************************
************************ identical cases *****************************
**********************************************************************
```
[hercules.gnu.fix.fix.matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/14840765/hercules.gnu.fix.fix.matrixCompSummary.txt)
[hercules.gnu.fix.fix.matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/14840764/hercules.gnu.fix.fix.matrixCompFull.txt)
matrixDiff.txt was too big to include (~25GB!)


Approved.

**Comment by MatthewMasarik-NOAA** [comment]
Thanks @JessicaMeixner-NOAA, adding a new platform and the gnu compiler gives us more options for validating PRs at NCEP.

**Pull request merged**
